### DISCUSSION TITLE
Windows builds failing

### DISCUSSION THREAD

**Discussion by sfiannaca** [Mobile-Artificial-Intelligence/maid#35] [Ideas]
Hi @danemadsen I don't know much about about C++ or C, but I made an attempt, and I managed to manually execute the Windows build. 
Maybe my little experience could be useful for troubleshooting

Steps I followed:
1. Delete the 'build' folder.

2. 'git checkout 67a38eb2'.

3. Checked this line: 
   Platform.isWindows ? 'butler.dll' : libbutler.so')); // android and linux

4. Run 'dart run msix:create'.

5. Now, within 'maid\build\windows\runner\Release', neither 'butler' nor 'llama' dlls were generated.

6. So, switch back to the main branch using 'git checkout f229036'.

7. Re-run 'dart run msix:create'.

8. The build failed, but the dlls were created in the wrong directory:
   - 'maid\build\windows\bin\Release'
     - llama.dll
     - ggml_shared.dll
   - 'maid\build\windows\butler\Release'
     - butler.dll

9. Simply copy these three dlls to 'maid\build\windows\runner\Release' and execute 'maid.exe'

**Comment by danemadsen** [comment]
> Hi @danemadsen I don't know much about about C++ or C, but I made an attempt, and I managed to manually execute the Windows build. 
> Maybe my little experience could be useful for troubleshooting
> 
> 
> Steps I followed:
> 1. Delete the 'build' folder.
> 
> 2. 'git checkout 67a38eb2'.
> 
> 3. Checked this line: 
>    Platform.isWindows ? 'butler.dll' : libbutler.so')); // android and linux
> 
> 4. Run 'dart run msix:create'.
> 
> 5. Now, within 'maid\build\windows\runner\Release', neither 'butler' nor 'llama' dlls were generated.
> 
> 6. So, switch back to the main branch using 'git checkout f229036'.
> 
> 7. Re-run 'dart run msix:create'.
> 
> 8. The build failed, but the dlls were created in the wrong directory:
>    - 'maid\build\windows\bin\Release'
>      - llama.dll
>      - ggml_shared.dll
>    - 'maid\build\windows\butler\Release'
>      - butler.dll
> 
> 9. Simply copy these three dlls to 'maid\build\windows\runner\Release' and execute 'maid.exe'

Yeah that's basically the right way to build windows ATM. The problem is I need the DLL's to be generated as part of the maids build process, not a separate build process. 

It needs to be this way so I can set up a github action that build windows the same way linux and android are now.

Getting windows working right now is currently a high priority for me but honestly it has me stumpt. 

If you arnt familiar with c/c++ i wouldnt recommend you attempt fixing it, Ive already tried contracting people on fiver who are c/c++ experts but nobody knows how to fix it.

**Reply by sfiannaca** [reply]
Gotcha

**Comment by danemadsen** [comment]
Ok yeah i fixed it. Turned out to be a bug with CMake
